File: Scripts/Player/PlayerStateMachine/PlayerStates/PlayerMove.cs
using UnityEngine;

public class PlayerMove : IPlayerState
{
    /// <summary>Determines if player state needs to change states and calls Move method</summary>
    /// <param name="player">Instance of player class</param>
    /// <returns>Returns IPlayerState interface</returns>
    public IPlayerState DoState(Player player)
    {
        Move(player);
        if (Input.GetKey(KeyCode.Space) && player.nextDodge < Time.time)
            return player.playerDodge;
        if (Input.GetAxisRaw("Horizontal") != 0 || Input.GetAxisRaw("Vertical") != 0)
            return player.playerMove;
        return player.playerIdle;
    }

    /// <summary>
    /// Gets direction that player needs to move from keyboard inputs
    /// </summary>
    /// <param name="player">Instance of player class</param>
    private void Move(Player player)
    {
        player.moveDirection.x = Input.GetAxisRaw("Horizontal");
        player.moveDirection.y = Input.GetAxisRaw("Vertical");
        
        player.moveDirection.Normalize();
    }
}

File: Scripts/Player/PlayerStateMachine/PlayerStates/player_move.py
import time

import pygame

from Scripts.Player.PlayerStateMachine.player_state import PlayerState


class PlayerMove(PlayerState):

    def __init__(self):
        super().__init__(None)

    def do_state(self, player):
        """Determines if player state needs to change states and calls Move method
        :param player: Instance of player class
        :returns: Returns PlayerState interface
        """

        # Get the state of all keys
        keys = pygame.key.get_pressed()

        # Update player position based on key input
        if any([keys[pygame.K_w], keys[pygame.K_s], keys[pygame.K_a], keys[pygame.K_d]]):
            return player.playerMove

        if keys[pygame.K_SPACE] and player.next_dodge < time.time():
            return player.player_dodge

        return player.playerIdle

    def move(self, player):
        """
        Gets direction that player needs to move from keyboard inputs
        :param player: Instance of player class
        """
        # Update player position based on key input
        # if keys[pygame.K_w]:
        #     movement_input.y = -1
        # if keys[pygame.K_s]:
        #     movement_input.y = 1
        # if keys[pygame.K_a]:
        #     movement_input.x = -1
        # if keys[pygame.K_d]:
        #     movement_input.x = 1
        #
        # player.moveDirection.x = Input.GetAxisRaw("Horizontal");
        # player.moveDirection.y = Input.GetAxisRaw("Vertical");
        #
        # player.moveDirection.Normalize();
        ...
